src/lib/stepper/stepper.ts

```diff
@@ -131,7 +135,13 @@ export class MatStepper extends _CdkStepper implements AfterContentInit {
   encapsulation: ViewEncapsulation.None,
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class MatHorizontalStepper extends MatStepper { }
+export class MatHorizontalStepper extends MatStepper {
+  /** Whether the label should display in the alternative or normal mode. */
+  @Input()
+  get alternativeLabel(): boolean { return this._alternativeLabel; }
```

Comment:
Although the spec calls it an "alternative label", I don't think it's an intuitive name to give it in the API. How about `labelPosition` which can have the following values: `bottom` - below the icon, `end` - after the icon (to the right in LTR / to the left in RTL). This would also leave open the possibility of future values like `top` and `start`

Reply:
How should I change CSS classes name with the API you proposed?
label-bottom and label-end don't really seem good names...
I agree with you that `alternativeLabel` may not be the perfect name, but I'd prefer leaving it as an on/off property until someone actually needs to put it in top or start position...

(Sorry for delay, I was in vacation)

Reply:
I'm not saying we should add support for `start` and `top` right now. I'm just saying that its more clear what the API does if we call it `labelPosition` and it leaves us open to adding new values in the future. I think class names like `.mat-step-label-position-end` would be good

Reply:
I should have addressed all of yor concerns.
I did not changed previous code (I did not created a `.mat-step-label-position-end` SCSS class) to avoid breaking things (even if that class is actually added to the element for consistency).

Now the alternative label is applied when `labelPosition` is set to `bottom`